fml/message_loop_impl.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/message_loop_impl.h"
+#include "lib/ftl/build_config.h"
+
+#if OS_MACOSX
+
+#include "flutter/fml/platform/darwin/message_loop_darwin.h"
+using PlatformMessageLoopImpl = fml::MessageLoopDarwin;
+
+#else
+
+#error This platform does not have a message loop implementation.
+
+#endif
+
+namespace fml {
+
+ftl::RefPtr<MessageLoopImpl> MessageLoopImpl::Create() {
+  return ftl::MakeRefCounted<::PlatformMessageLoopImpl>();
+}
+
+MessageLoopImpl::MessageLoopImpl() : order_(0) {}
+
+MessageLoopImpl::~MessageLoopImpl() = default;
+
+void MessageLoopImpl::PostTask(ftl::Closure task, ftl::TimePoint target_time) {
```

Comment:
> Or just drop it on the floor?

Just dropping it on the floor is sufficient to run its destructor.

> What if it is a delayed task?

Those too, but let's remove the concept of delayed task.

> It seems odd that there can be cases where task execution happens on a thread and time that is different from the one the caller expects.

Yes, we definitely don't want to run the task in this case.  We just want to drop it on the floor in order to run its destructor.